### ISSUE TITLE
Assess all Ethereum-specific code in agents and refactor as necessary

### ISSUE THREAD

**Issue by tkporter** [hyperlane-xyz/hyperlane-monorepo#1283]
* Blocked on v2 agent PRs being merged (https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/1227 https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/1280)
* Suggested steps:
1. Add a new dummy field to the `enum ChainConf` -- what breaks? If it's just a matter of adding a new branch to a match statement that takes in non-Ethereum-specific variables, that's okay. If it's something more structural where the variables that the match statement has access to are themselves Ethereum-specific, we should try to fix this
2. Search the entire repo for `ethers` imports. Types like H256 and U256 are likely to be pretty pervasive, which is fine for situations where we use them to deal with 32 byte blobs (like the sender or receiver of a message, etc), or when we want big number support. However, we should scrutinize other usage of ethers types -- e.g. are we using `H160` / `Address` in places that aren't Ethereum-specific? Hash functions like keccak or ECDSA signature handling are also probably fine to continue using ethers for.
3. If there are large parts of the codebase that may require a lot of changes, for now just take note of these and create an issue (e.g. say if the scraper has some Eth-specific transaction assumptions, it's probably better to just add scraper support later on)
4. Take note of where code changes will be required - does it generally seem like it's limited to converting settings into contract abstractions, and to add a new crate in `rust/chains`?

**Comment by mattiecnvr** [comment]
1) Actually nothing looks to major on this front, only two places "broke" but not in a way that indicates a fundamental design problem but just that a new variant was needed. The return types of the functions did support dynamic returns so supporting a new chain should not cause issues.

**Comment by mattiecnvr** [comment]
3) It is kinda difficult to look for issues preemptively but a few things that I think will need some work
    * metrics will need to be extended, possibly be reworked
    * middleware stack might need to get ported, e.g. retrying and quorum logic might be nice to have for other non-evm chains
    * `HyperlaneProvider` and the associated return types might need to get updated
    * scraper database schemas might need to get updated
    * Helm charts and deployment configs will probably need to be updated to support multiple config types
    * Probably more
